src/ui.rs

```diff
@@ -370,107 +370,39 @@ impl<'a> UiFrame<'a> {
-                                                    }
-
-                                                    combo_token.end(ui);
-                                                }
-
-                                                result
-                                            };
-
-                                            if combo_changed {
-                                                if let Some(selected_var_index) = selected_var_index
-                                                {
-                                                    change = Some((
-                                                        stmt_index,
-                                                        arg_index,
-                                                        ast::Expr::Var(ast::VarExpr::new(
-                                                            visible_vars[selected_var_index],
-                                                        )),
-                                                    ))
-                                                } else {
-                                                    change = Some((
-                                                        stmt_index,
-                                                        arg_index,
-                                                        ast::Expr::Lit(ast::LitExpr::Nil),
-                                                    ))
-                                                }
+                                            let changed_expr = draw_var_combo_box(
+                                                ui,
+                                                session,
+                                                stmt_index,
+                                                arg,
+                                                Ty::Geometry,
+                                                &input_label,
+                                            );
+
+                                            if let Some(changed_expr) = changed_expr {
+                                                change = Some((
+                                                    stmt_index,
+                                                    arg_index,
+                                                    changed_expr,
+                                                ));
+                                            }
+                                        }
+                                        ParamRefinement::GeometryArray => {
+                                            let changed_expr = draw_var_combo_box(
+                                                ui,
+                                                session,
+                                                stmt_index,
+                                                arg,
+                                                Ty::GeometryArray,
+                                                &input_label,
```

Comment:
Not sure if this is the right way and place, but what about wrapping the group name into square brackets here to distinguish the groups from meshes? Something like `&fmt!("[{}]", input_label)`

Reply:
How nerdy of you. I like it! :+1: